sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -65,13 +66,16 @@
 
     private TableAsyncClient(String tableName, AzureTableImpl implementation, SerializerAdapter serializerAdapter) {
         this.serializerAdapter = serializerAdapter;
+
         try {
-            if (tableName == null || tableName.isEmpty()) {
-                throw new IllegalArgumentException("'tableName' must be provided to create a TableClient");
+            if (isNullOrEmpty(tableName)) {
+                throw new IllegalArgumentException("'tableName' must be provided to create a TableClient.");
```

Comment:
We have guidelines to throw NPE when the input is null and IllegalArgumentException if the input is non-null but invalid. Update other methods in this class as well.
https://azure.github.io/azure-sdk/java_introduction.html#java-errors-system-errors